Sources/SwiftMemcache/MemcachedConnection.swift

```diff
@@ -0,0 +1,90 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-memcache-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-memcache-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-memcache-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import NIOPosix
+
+public actor MemcachedConnection {
+    private let channel: Channel
+    
+    public init(host: String, port: Int) async throws {
+        let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        let bootstrap = ClientBootstrap(group: group)
+            .channelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
+            .channelInitializer { channel in
+                let promise = channel.eventLoop.makePromise(of: MemcachedResponse.self)
+                let responseHandler = ResponseHandler(p: promise)
+                return channel.pipeline.addHandlers([MessageToByteHandler(MemcachedRequestEncoder()), ByteToMessageHandler(MemcachedResponseDecoder()), responseHandler])
+            }
+
+        self.channel = try await bootstrap.connect(host: host, port: port).get()
```

Comment:
We should use the new `AsyncChannel` SPI. To get those add `@_spi(AsyncChannel)` before both imports at the top of file. Then use the new `connect` methods that return you a `NIOAsyncChannel`. For this you have to pass the `inbound` and `outbound` types to the `connect` method.

Reply:
You can then use the `NIOAsyncChannel.outboundWriter` and `NIOAsyncChannel.inboundStream` to send requests and wait for responses.